### PR TITLE
Fix primary disk resize on clone and add tests

### PR DESCRIPTION
**Pull request by sylviamoss**
Closes https://github.com/hashicorp/packer/issues/10770

### CODE REVIEW

File: builder/vsphere/driver/vm.go [hunk 1]
```diff
@@ -597,35 +607,25 @@ func (vm *VirtualMachineDriver) Customize(spec types.CustomizationSpec) error {
...
-	}
-
-	task, err := vm.vm.Reconfigure(vm.driver.ctx, confSpec)
-	if err != nil {
-		return err
-	}
-
-	_, err = task.WaitForResult(vm.driver.ctx, nil)
-	return err
+	}, nil
```
Comment (azr) [inline builder/vsphere/driver/vm.go:628]: Nice one !

File: builder/vsphere/clone/step_clone_test.go [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+//	state.Remove(multistep.StateCancelled)
+//
+//	// Should not destroy if VM is not set
+//	state.Remove("vm")
+//	state.Put(multistep.StateCancelled, true)
+//	step.Cleanup(state)
+//	if vm.DestroyCalled {
+//		t.Fatalf("vm.Destroy should not be called")
+//	}
+//}
```
Comment (azr) [inline builder/vsphere/clone/step_clone_test.go:289]: Should we keep these commented ?

**Review by azr — APPROVED** [review]
Nice one, LGTM !

File: builder/vsphere/clone/step_clone_test.go (hunk 2 above)
Comment (sylviamoss) [reply builder/vsphere/clone/step_clone_test.go:289]: No!! I forgot to remove them. hehe

**Review by nywilken — APPROVED** [review]
LGTM

**Pull request merged**